packages/devtools_app/lib/src/screens/profiler/cpu_profile_model.dart

```diff
@@ -168,24 +168,25 @@ class CpuProfileData {
     _cpuProfileRoot = CpuStackFrame.root(profileMetaData);
   }
 
-  factory CpuProfileData.parse(Map<String, dynamic> json) {
+  factory CpuProfileData.parse(Map<String, dynamic> json_) {
+    final json = _CpuProfileDataJson(json_);
```

Comment:
I wonder if we can go a step further and just have a single CpuProfileData class that is is an extension type on Map<String, dynamic>
?
profileMetaData and individual frames could be computed as needed from the json. This might be horrible for performance as some of the logic for computing frames looks a bit expensive.